src/components/imageApprovalModal/index.tsx

```diff
@@ -0,0 +1,226 @@
+
+interface ImageApprovalModal {
+  visible: boolean;
+  onClose: () => void;
+  tableData: FilterImageTableData | null;
+  approvedOrRejectedImageIds: number[];
+  setApprovedOrRejectedImageIds: Dispatch<SetStateAction<number[]>>;
+  allData: FilterImageTableData[];
+  setSelectedImage: Dispatch<SetStateAction<FilterImageTableData | null>>;
+}
+
+const ImageApprovalModal: React.FC<ImageApprovalModal> = (props) => {
+  const [open, setOpen] = useState(props.visible);
+  const [data, setData] = useState(props.tableData);
+  const [isRejectionTextOpen, setIsRejectionTextOpen] = useState(false);
+  const [rejectionReason, setRejectionReason] = useState('');
+  const [allData, setAllData] = useState(props.allData);
+
+  const close = () => {
+    setOpen(false);
+    props.onClose();
+  };
+
+  const openRejectionTextBox = () => {
+    setIsRejectionTextOpen(true);
+  };
+
+  const handleNextSubmission = () => {
+    const currentIndex = allData.findIndex(
+      (siteData) => siteData.key === data?.key,
+    );
+    const nextIndex = (currentIndex + 1) % allData.length;
+    const nextImage = allData[nextIndex];
+    setData(nextImage);
+  };
+
+  const footer = (
+    <Space style={{ width: '100%', justifyContent: 'space-between' }}>
+      <Button style={{ color: 'gray' }} onClick={close}>
+        <LeftOutlined /> Back to List
+      </Button>
+      <Button style={{ color: 'gray' }} onClick={handleNextSubmission}>
+        Next Submission <RightOutlined />
+      </Button>
+    </Space>
+  );
+
+  const rejectionReasonStyle = {
+    background: 'rgb(239, 239, 239)',
+    borderRadius: '5px',
+    marginTop: '5%',
+    padding: '3%',
+    paddingRight: '10%',
+  };
+
+  const StatusHeader = () => {
+    return (
+      <>
+        <h1 style={{ color: 'black', marginBottom: '-8px' }}>Status</h1>
+        <div
+          style={{
+            display: 'flex',
+            width: '100%',
+            backgroundColor: 'rgb(232, 240, 215)',
+            padding: '2%',
+            borderRadius: '5px',
+          }}
+        >
+          <Button
+            type="primary"
+            style={{ flex: 1, marginRight: '8px', color: 'black' }}
+            onClick={onClickAccept}
+          >
+            Accept
+          </Button>
+          <Button
+            type="primary"
+            style={{ flex: 1, color: 'black' }}
+            onClick={openRejectionTextBox}
+          >
+            Reject
+          </Button>
+        </div>
+      </>
+    );
+  };
+
+  if (!data) {
+    return null;
+  }
+
+  const treeSummaryTextStyle: CSSProperties = {
+    whiteSpace: 'nowrap', // Prevent text from wrapping
+  };
+
+  async function onClickReject() {
+    const toReject: Promise<void>[] = [];
+    if (!data) {
+      return null;
+    }
+    toReject.push(protectedApiClient.rejectImage(data.key, rejectionReason));
+    Promise.all(toReject).then(() => {
+      props.setApprovedOrRejectedImageIds((prevIds) => [...prevIds, data.key]);
+      close();
```

Comment:
⛏️ Can we add some feedback with the `message` antd component (specifically, `message.success()` on success, and `message,error()` on failure) once the API call finishes? That way, the user can know for sure the outcome of their action instead of having to assume  it went through fine.